File: solutions/ruby/raindrops/2/raindrops.rb
module Raindrops

  def self.convert(number)
    sounds = {
      3 => 'Pling',
      5 => 'Plang',
      7 => 'Plong',
    }

    droplet_sounds = sounds.select { |divisor, _| (number % divisor).zero? }
    droplet_sounds.empty? ? number.to_s : droplet_sounds.values.join
  end

end

File: solutions/python/raindrops/2/raindrops.py
def convert(num):
    raindrops = ''

    if num % 3 == 0:
        raindrops += 'Pling'

    if num % 5 == 0:
        raindrops += 'Plang'

    if num % 7 == 0:
        raindrops += 'Plong'

    return raindrops or str(num)

